app/controllers/sign_up/select_email_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module SignUp
+  class SelectEmailController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    def show
+      @sp_name = sp_name
+      @user_emails = user_emails
+      @last_sign_in_email_address = EmailContext.new(current_user).last_sign_in_email_address.email
+      @select_email_form = new_select_email_form
+    end
+
+    def create
+      @select_email_form = new_select_email_form
+
+      result = @select_email_form.submit(form_params)
+      if result.success?
+
+        EmailAddress.update_last_sign_in_at_on_user_id_and_email(
```

Comment:
This is a pretty clever solution to set the last sign-in to make the current logic for deciding which email is shared use this email. But I'm not sure that helps provide the sort of value I'd expect with this feature allowing a user to select an email _and have that email be remembered the next time the user signs in_. Since the consent screen is only shown once, we really only have this one opportunity.

What I had in mind was something more like a new column on the `identities` table to map remember the selected email, and update the logic for deciding which email is shared to use that if it's available.

Reply:
I have made changes to include database permanency for the updated email.

Reply:
If we're storing on the identity record, I'm not sure I'd expect to see this code where we're updating the last sign-in on email? Although I guess that's an interesting question. I'm not sure the repercussions of this.